### PR TITLE
Dark mode

### LINKED ISSUE #10156: Dark Mode 😎
### User Story

As a user I want to enable a dark mode setting to make the interface easier on the eyes in dim light conditions, preserve battery on OLED displays, or as a personal preference. 

### Description

Scope of this issue is as follows:

1. We place an **Appearance** setting in profile

Frame 25

---

2. Inside we allow selecting a preference, to **always use Light Mode**, **always use Dark Mode**, or **follow the system setting**.

Group 1

---

3. We add Dark Theme specific colour set that maps to Light, and allow our components to be rendered using either one of them

Screenshot 2020-03-12 at 09 19 01

Group 21

Group 22

Group 23

Group 24

---

### Acceptance Criteria

I'm allowed to select Dark Theme as my appearance preference and see the interface change like so

Group 89

Group 90

Group 91

Group 92

Group 93

### Notes

[Designs in Figma](https://www.figma.com/file/9hkPcxsqDTWa9b9c49OX3X/Dark-Mode?node-id=1%3A85)

some of the dark theme colours are subject to change in the next day or two while I do some testing 😎 
when in need of assistance, holla at maciej.stateofus.eth

Thank you, u r awesome! 🙏

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #10156

set colors
https://github.com/status-im/status-react/pull/10171/files#diff-5599188f73da96ec5d301f9ae92ea4eeR98

re-rendering app 
https://github.com/status-im/status-react/pull/10171/files#diff-8d6f8805fd8841875fc7095d8504cf23R110

### CODE REVIEW

**Comment by errorists** [comment]
<3 
for QA, original scope of this work is here https://github.com/status-im/status-react/issues/10156

File: ios/launch-image-universal.storyboard [hunk 1]
```diff
@@ -32,6 +32,6 @@
         </scene>
     </scenes>
     <resources>
-        <image name="launch_image" width="90" height="101"/>
+        <image name="launch_image" width="164" height="169"/>
```
Comment (errorists) [inline ios/launch-image-universal.storyboard]: hey I noticed this here, is that the Status logo that gets shown on splash screen? why is it not a square 1:1 format, different width - height?

Comment (flexsurfer) [reply ios/launch-image-universal.storyboard]: dunno how it works, xcode always generates lots of magic staff, but the logo looks good on splash screen

**Comment by errorists** [comment]
Issues:

- [x] Tab Bar should use dark theme specific shadow colour and opacity values
- [x] Same for tooltips^^
- [x] Same for accounts in Account Explorer ^^
- [x] Same for floating buttons
- [x] Status account in Account Explorer should continue to use the same colour regardless if it's Light Theme or not
- [x] Chat headers ignore theme setting
- [x] Notification badges inside list items use wrong text colour, should be white for both light and dark theme
- [x] Radio buttons inside list items use wrong pill colour, should be white for both light and dark theme
- [x] Outgoing messages, all text styles should remain unchanged, currently timestamps, quoted text snippets, quote separation are dark. Outgoing blue messages should remain unchanged between light and dark theme. 
- [x] Composing replies, quoted text is black
- [x] Destructive list items (the ones that are red), the light background below icon should be a 10% Red
- [x] Network settings, selected list item, in dark theme the icon remains black, should be white
- [ ] New Group Chat screens are messed up
- [ ] Changing the Appearance setting causes UI 'restart', taking me back to Chat tab

will edit and continue as I test more

**Comment by jakubgs** [comment]
I tried the PR build and it looks fine, but bottom bar and top bar in ANdroid are white.

What's weird is that the notifications bar doesn't show the notification icons, or time.

Screenshot_20200323-163123

**Comment by flexsurfer** [comment]
@jakubgs thanks, can repro, will fix soon

**Comment by flexsurfer** [comment]
@jakubgs could you please try build #18, the status bar should be fixed, but not sure about the bottom if it's still white pls tell the model of your phone, thanks!

**Comment by flexsurfer** [comment]
@errorists all your points fixed, group chats not in the scope of this PR, because of UI in progress in another PRs, `UI 'restart', taking me back to Chat tab` I'll try to fix this

**Comment by errorists** [comment]
thanks, that was fast 💪 !! let's wait for QA now, I didn't test things like chat commands, transactions etc

**Comment by jakubgs** [comment]
The newest build has no white bars! Thanks Andrey, amazing work.
[Screenshot_20200324-150913](https://user-images.githubusercontent.com/2212681/77434677-84b8e380-6de1-11ea-9107-f8d88b9eeda1.jpg)

**Review by Ferossgp — APPROVED** [review]
Good work! I would name colors differently like `primary-color`, `text-color` etc. based on not how it looks on light mode but on the purpose it has. That's because it helps me to build a better mental model. When looking at a component code I can understand how it looks on each theme and reading color/white confuses. An extra thing that should be researched is that we do not have some components which are themed by the react-navigation https://reactnavigation.org/docs/themes (Seems like we do not use any, but we need to have that in mind)

**Comment by errorists** [comment]
@Ferossgp makes a good point about colours, unfortunately I couldn't make semantic naming work in design because for example, take black, we use it both for primary text but also icons, it's even more difficult with greys since they're used all over the place

**Comment by errorists** [comment]
dude it's sooo good! tested, all of the issues I had are confirmed fixed. 🙇 great work 🙇‍♀

**Comment by errorists** [comment]
one more thing I'm not even sure it's possible, for security reasons we blank the screen on iOS when multitasking. would be cool if it was black too

IMG_1383

**Comment by flexsurfer** [comment]
@errorists done

**Comment by errorists** [comment]
legend

**Comment by churik** [comment]
@flexsurfer 
"Sync All" button is overlapped in PR build on Android and not visible on IOS (sync settings > Devices)
sync_all_devices
This is the reason of failed `test_pair_devices_sync_one_to_one_contacts_public_chat`.
Shortly I'll push fix for remaining tests.

**Comment by flexsurfer** [comment]
@jakubgs when build will be ready, could you try , I added a black splash screen for Android, but it could break bottom bar on your phone, thanks

**Comment by jakubgs** [comment]
Tested the most recent build and both the bottom bar and the [switching view](https://user-images.githubusercontent.com/2212681/77450005-f9484e00-6df2-11ea-965c-2fd0c9b2c601.jpg) look good.

**Comment by flexsurfer** [comment]
when "autosignin" white loading screen appears

**Comment by churik** [comment]
## IOS
### Appearance in Status - Dark, In Phone Settings - Light
#### 1) Regardless of Light Settings in Main mode system dialogs and keybord are dark in dark theme in other app (compared with telegram)
 
tg_st1

photo_2020-03-25 16 49 18 jpeg 2020-03-25 16-50-00

Also in this setup after force-logout (due to https://github.com/status-im/status-react/issues/10095) login screen is white.
The white screen is also at attempt to revoke touch-id.

### Appearance in Status - Dark, In Phone Settings - Dark
#### 2) Collectibles
photo_2020-03-25 16 57 03

#### 3) Add members to group chats + group info screens
photo_2020-03-25 16 57 47

#### 4) Currency screen in Wallet -> Settings
curr


#### 5) Top-bar notifications (not sure it should look like this)
photo_2020-03-25 17 01 01


#### 6) "Sync all" button is not shown on IOS (mentioned in https://github.com/status-im/status-react/pull/10171#issuecomment-603313191)

**Comment by errorists** [comment]
sweet lord, that Collectibles view, think I'm gonna puke 🤢🙈 

@churik on my phone those dialogs are using proper dark appearance styling

IMG_1385

**Comment by churik** [comment]
@errorists 
it is in case when main "display" setting in phone settings is Light.
In fact, I'm not sure it is a prio - but I attached screens from another app to compare with

**Comment by flexsurfer** [comment]
1) lets implement separately if needed (I would say low priority)
2) collectibles  **fixed**
3) group charts not in the scope of this PR
4) Currency - **fixed**
5) not in the scope of this PR
6) **fixed**

@churik thanks

**Review by churik — APPROVED** [review]

**Comment by churik** [comment]
Too hard to maintain PR up-to-date - decided to merge it and later continue dark mode testing on Android.

**Pull request merged**